app/scripts/services/applicationGenerator.js

```diff
@@ -402,6 +402,55 @@ angular.module("openshiftConsole")
       return service;
     };
 
+  // Public method for creating a persistent volume claim.
+    // Expects pvcOptions from the osc-persistent volume claim directive.
+    scope.createPersistentVolumeClaim = function(pvcOptions, serviceName, labels) {
+      return scope._generatePersistentVolumeClaim({
+        labels: labels || {},
+        params: angular.extend({ include: true }, pvcOptions.persistentVolumeClaimForm)
+      }, pvcOptions.persistentVolumeClaimForm.name, serviceName);
+    };
```

Comment:
If this can only be used with the persistent volume claim form from CreatePersistentVolumeClaimController, I'd probably just put the function in that controller rather than ApplicationGenerator.